### ISSUE TITLE
AWS::CloudFormation::Stack TemplateBody property not recognized

### ISSUE THREAD

**Issue by sparr** [aws-cloudformation/cfn-lint#2975]
### CloudFormation Lint Version

0.83.4

### What operating system are you using?

MacOS 14.1

### Describe the bug

An `AWS::CloudFormation::Stack` resource with a TemplateBody property produces two errors. One E3003 for the missing required TemplateUrl property, one E3002 for the unrecognized TemplateBody property.

### Expected behavior

TemplateBody should be recognized as a valid alternative to TemplateUrl

### Reproduction template

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  TestStack:
    Type: AWS::CloudFormation::Stack
    Properties:
      TemplateBody:
        AWSTemplateFormatVersion: '2010-09-09'
```

**Comment by kddejong** [comment]
From the spec.  Looks like we are going to need to patch the spec.  There is a lot missing there.
```
"AWS::CloudFormation::Stack": {
      "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-stack.html",
      "Properties": {
        "NotificationARNs": {
          "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-stack.html#cfn-cloudformation-stack-notificationarns",
          "DuplicatesAllowed": false,
          "PrimitiveItemType": "String",
          "Required": false,
          "Type": "List",
          "UpdateType": "Mutable"
        },
        "Parameters": {
          "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-stack.html#cfn-cloudformation-stack-parameters",
          "DuplicatesAllowed": false,
          "PrimitiveItemType": "String",
          "Required": false,
          "Type": "Map",
          "UpdateType": "Mutable"
        },
        "Tags": {
          "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-stack.html#cfn-cloudformation-stack-tags",
          "DuplicatesAllowed": true,
          "ItemType": "Tag",
          "Required": false,
          "Type": "List",
          "UpdateType": "Mutable"
        },
        "TemplateURL": {
          "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-stack.html#cfn-cloudformation-stack-templateurl",
          "PrimitiveType": "String",
          "Required": true,
          "UpdateType": "Mutable"
        },
        "TimeoutInMinutes": {
          "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-stack.html#cfn-cloudformation-stack-timeoutinminutes",
          "PrimitiveType": "Integer",
          "Required": false,
          "UpdateType": "Mutable"
        }
      }
    },
```

**Comment by kddejong** [comment]
My bad.  I messed this up and will be undoing that PR.

**Comment by kddejong** [comment]
In the docs this is stated
```
A subset of AWS::CloudFormation::Stack resource type properties listed below are available to customers using AWS CloudFormation, AWS CDK, and AWS Cloud Control API to configure.

    NotificationARNs

    Parameters

    Tags

    TemplateURL

    TimeoutInMinutes
```

When using TemplateBody you will get `Encountered unsupported property TemplateBody`

**Comment by sparr** [comment]
> the below properties are set by the parent stack
> ...
> TemplateBody

In the example given at the top of this issue, it is being set by the parent stack, and is thus valid.

Notably, `aws cloudformation validate-template` parses it as valid.

**Comment by kddejong** [comment]
I noticed the validation didn't work right either. I believe this is because the schema says its a valid property but it seems that we fail it later on in the deployment process.

**Comment by sparr** [comment]
I stand corrected. Despite validate-template accepting it, create-stack fails with `Encountered unsupported property TemplateBody`. I will open an issue with AWS about validate-template misbehaving here, and continue trying to figure out how to use nested stack templates.

**Comment by kddejong** [comment]
thanks @sparr .  I think the big thing here is you have to use TemplateURL.  The CLI tools will help upload nested stacks and put in the S3 URL for deployment. 

I would raise support requests for adding support for TemplateBody.

**Comment by sparr** [comment]
https://github.com/aws/aws-cli/issues/8412
